device/delta/x86_64-delta_ag5648-r0/plugins/psuutil.py

```diff
@@ -45,8 +46,8 @@ def get_psu_presence(self, index):
         Base_bus_number = 49
         status = 0
         try:
-            p = os.popen(self.psu_presence.format(index + Base_bus_number) + "> /dev/null 2>&1")
-            if p.readline() != None:
```

Comment:
>index + Base_bus_number) [](http://example.com/codeflow?start=64&length=24)

I believe original code should be fixed with `hex(index + Base_bus_number)`. Since normally there will be only 2 PSU, it does not matter significantly.

`hex()` will make your new args manipulation easier. No need to append strings.

Reply:
In your new code, you will have "0x0x".

Reply:
I updated to original code.